src/nupic/algorithms/TemporalMemory.hpp

```diff
@@ -512,6 +512,7 @@ using nupic::algorithms::connections::Synapse;
 
 public:
   Connections connections;
```

Comment:
can you please make public only a `const Connections&`, and use private `Connections connections_` ?